releasenotes/notes/curve-analysis-02a702a81e014adf.yaml

```diff
@@ -0,0 +1,51 @@
+---
+prelude: >
+    To be written.
+features:
+  - |
+    :class:`CurveAnalysis` delegates the core fitting functionality to the LMFIT package.
+    Analysis class author can flexibly define new analysis instance
+    with LMFIT ``Model`` objects. In addition, new analysis option ``fit_method`` is added
+    so that user can perform fit with several algorithms offered by the LMFIT minimizer.
+    See LMFIT documentation for user guide.
+  - |
+    New dataclass :class:`.CurveFitResult` is introduced. This dataclass includes
+    rich context of curve fitting status with several extra statistics.
+    This is a wrapper class of LMFIT ``MinimizerResult`` with some extention.
```

Comment:
Its isn't really a wrapper, it's a minimal dataclass based on the result properties of MinimizerResult